kratos/future/input_output/vtu_output.cpp

```diff
@@ -0,0 +1,1689 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <algorithm>
+#include <filesystem>
+#include <numeric>
+#include <type_traits>
+
+// External includes
+
+// Project includes
+#include "includes/data_communicator.h"
+#include "input_output/base_64_encoded_output.h"
+#include "input_output/vtk_definitions.h"
+#include "tensor_adaptors/flags_tensor_adaptor.h"
+#include "tensor_adaptors/historical_variable_tensor_adaptor.h"
+#include "tensor_adaptors/node_position_tensor_adaptor.h"
+#include "tensor_adaptors/variable_tensor_adaptor.h"
+#include "utilities/container_io_utils.h"
+#include "utilities/data_type_traits.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+#include "utilities/string_utilities.h"
+#include "future/utilities/xml_utilities/xml_appended_data_element_wrapper.h"
+#include "future/utilities/xml_utilities/xml_elements_array.h"
+#include "future/utilities/xml_utilities/xml_in_place_data_element_wrapper.h"
+#include "future/utilities/xml_utilities/xml_utils.h"
+
+// Include base h
+#include "vtu_output.h"
+
+namespace Kratos::Future {
+
+namespace {
+
+std::string GetEndianness()
+{
+    int i = 0x0001;
+
+    if (*reinterpret_cast<char*>(&i) != 0) {
+        return "LittleEndian";
+    } else {
+        return "BigEndian";
+    }
+}
+
+template<class... T>
```

Comment:
why do we need variadic templates here? i am pretty convinced you could avoid that simply calling twice the function with different arguments ...

Reply:
removed, and simplified. :D